### ISSUE TITLE
Layer switcher obscured in Neatline 2

### ISSUE THREAD

**Issue by rachaelking** [scholarslab/Neatline#233]
With Neatline Waypoints installed, it's no longer possible to access the layer switcher -- it's under the list of items on the right-hand side. This is a problem for projects that may have multiple map layers to provide comparison.

**Comment by davidmcclure** [comment]
Yep, I see what you're talking about - that is a problem. Should be simple to fix - just need to add a right offset to the layer switcher when both the Waypoints and Simile widgets are enabled. I'll try to get this into the next point release. 

And sorry for the slow response - we were all off at a conference last week, and I didn't notice this until just now!

**Comment by davidmcclure** [comment]
Hey Rachel - Actually, on the most recent builds of Neatline, I can't reproduce this - the layer switcher has a `z-index` that puts it on top of the waypoints container. Two questions:
- What version of Neatline are you using?
- What browser?

Thanks!

**Comment by rachaelking** [comment]
Hey David -- sorry to be slow responding. I'm using Neatline 2.0.0 and Omeka 2.0.4. It looks the same in both Chrome and Safari. Thanks!

**Comment by lmullen** [comment]
I have the same problem, but it doesn't appear to be related to the Waypoints plugin.

I'm using Neatline 2.2.1. When I add a WMS layer from Geoserver, the historical map layer appears, but the underlying layer (Google Maps etc) does not. There is not a layer switcher icon visible, and as far as I can tell it's not present in the DOM. And when there is a historical layer the points on the map don't appear. Is this the same bug, or am I doing something wrong?

**Comment by davidmcclure** [comment]
Hey @lmullen, you're talking about when you add a "global" WMS layer in the exhibit add/edit form, is that right? When you do that, Neatline uses that WMS layer as the _only_ layer in the exhibit (that's why the layer switcher isn't displayed). This feature is designed to make it possible to use the WMS protocol as a way to build exhibits on top of really high-resolution imagery that would choke up the browser if you tried to load it all in at once as a static image. To put WMS layers on top of the regular spatial base layers (the more common use pattern), create records in an exhibit for the layers and fill in the WMS Address / WMS Layers fields under the "Style" tab. We could do a better job explaining this in the field descriptions in the exhibit form.

As for the disappearing points - what's probably going on is that the points were created when the WMS layer was set as the "global" base layer, in which case OpenLayers will use the spatial projection of the WMS layer (probably EPSG:4326). Then, when you change the base layer to one of the Google/OpenStreetMap/Stamen layers, which are EPSG:900913, the points aren't positioned correctly against the new coordinate system. (I bet you see what looks like a single point down under the horn of Africa - that's all the points, clustered around the "null island" at the origin of the spatial grid.)

@waynegraham, is there a way that we can handle this difference at the application level, so that people don't have this problem? Can we somehow force all layers into either EPSG:4326 or EPSG:900913? I have some memory of talking about this a couple years ago, and thinking there was some kind of significant obstacle, but I don't remember clearly.

**Comment by davidmcclure** [comment]
Also, @rachaelking, just to follow up about the original question, very belatedly - I'm perplexed by this! I just checked it out again with Neatline 2.2.1, and still can't replicate the problem - I see the layer switcher on top of the Waypoints containers, with the z-indexes at 1005 and 999 respectively. Are you on Windows, by any chance? I wonder if this some kind of weird CSS quirk related to the platform?

**Comment by lmullen** [comment]
Thanks, @davidmcclure. That's exactly what I needed to know: I hadn't figured out that I needed to create items for each historical layer.

**Comment by rachaelking** [comment]
That's odd -- not sure why I'm getting this. Not using Windows, but I
haven't updated to the most recent Neatline (I'm using 2.0.0, I think). I
will try updating and see what happens.

Rachael

On Mon, Mar 10, 2014 at 6:29 PM, davidmcclure notifications@github.comwrote:

> Also, @rachaelking https://github.com/rachaelking, just to follow up
> about the original question, very belatedly - I'm perplexed by this! I just
> checked it out again with Neatline 2.2.1, and still can't replicate the
> problem - I see the layer switched on top of the Waypoints containers, with
> the z-indexes at 1005 and 999 respectively. Are you on Windows, by any
> chance? I wonder if this some kind of weird CSS quirk related to the
> platform?
> 
> ## 
> 
> Reply to this email directly or view it on GitHubhttps://github.com/scholarslab/Neatline/issues/233#issuecomment-37254632
> .

## 

Rachael Scarborough King
Ph.D. candidate, Department of English
New York University
E-mail: rachael.king@nyu.edu, rachaelsking@gmail.com
Mobile: (646) 508-4162

**Comment by rachaelking** [comment]
Hi @davidmcclure, belatedly getting back to this - now updated to 2.2.4, and still having the same problem (in the screenshot, you can see that it's gray under the Waypoints box). Any thoughts?
![screen shot 2014-06-30 at 3 04 02 pm](https://cloud.githubusercontent.com/assets/5003656/3434368/5196fb66-0089-11e4-9639-c4b7caa5b58f.png)
.

**Comment by davidmcclure** [comment]
Hey @rachaelking, any chance this project is accessible online? I bet I could figure out what's going on here if I could tinker around in the web inspector on an exhibit where the problem is showing up.

Thanks!
David

**Comment by rachaelking** [comment]
Yes, it's at theperiodicalworld.org. Thanks!

On Mon, Jun 30, 2014 at 5:22 PM, davidmcclure notifications@github.com
wrote:

> Hey @rachaelking https://github.com/rachaelking, any chance this
> project is accessible online? I bet I could figure out what's going on here
> if I could tinker around in the web inspector on an exhibit where the
> problem is showing up.
> 
> Thanks!
> David
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/scholarslab/Neatline/issues/233#issuecomment-47588988
> .

## 

Rachael Scarborough King
Ph.D. candidate, Department of English
New York University
E-mail: rachael.king@nyu.edu, rachaelsking@gmail.com
Mobile: (646) 508-4162

**Comment by davidmcclure** [comment]
Hey Rachel,

Ok, I finally figured this out - the reason I couldn't reproduce the problem was that I wasn't using a Google layer in my test exhibits. I don't totally understand this, but the gist of it is that the Google layers don't conform to the same set of standards that the other layers do, which means that OpenLayers has to do some weird gymnastics to display them. This involves making changes to the markup structure inside the exhibit, which in turn throws off the CSS that Neatline uses to display the layer switcher.

Instead of trying to fix this at the root, which would be very difficult, I did this - just bump the layer switcher over so that it's flush with the left edge of the waypoints container:

![layer-switcher](https://cloud.githubusercontent.com/assets/814168/3465289/74c15e94-0262-11e4-9341-2dfa1ac82cfd.png)

@waynegraham, want to cut off a little patch release to get this out in the open? It's about time anyway - develop is about 85 commits ahead of master.
